src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -138,6 +138,8 @@ protected void FinishDocumentSetup(Document doc, Lucene.Net.Search.ScoreDoc scor
                     using (_projectionStorageScope = _projectionStorageScope?.Start() ?? _projectionScope?.For(nameof(QueryTimingsScope.Names.Storage)))
                         doc = DirectGet(ref retrieverInput, lowerId, DocumentFields.All);
 
+                    FinishDocumentSetup(doc, retrieverInput.Score);
```

Comment:
Not sure if this should be here? Shouldn't this be inside 'GetProjectionFromDocumentInternal'? Which case sn't handled there? With immediateResult? I see that ReturnProjection there handles that.

Reply:
Because `GetProjectionFromDocumentInternal` will get the fields to pull.
However, the fields to pull may also include the `@spatial`, so we need to add the fields before we call to that function